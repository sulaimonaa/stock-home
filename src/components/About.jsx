import React from 'react';// eslint-disable-line no-undef, no-unused-vars
import Navigation from './OtherNav';

const About = () => {
    return (
        <>
        <Navigation />
        <div className="wrap">
            <h3>About The Project</h3>
            <p>This is a simple project that uses react hook and redux to fetch financial data from an open API - Free Stock Market.<br/> The home page shows the card to display financial statement list, the stock market, and about project. Credit to Nelson Sakwa on Behance for the UI layout</p>
            <p>In the financial list page, you have list of financial statements you can filter and get details of a statement by click on it. Please, note that the free subscription of the API gives limited access i.e there are some statment that will return no data.</p>
        </div>
        </>
    );
}

export default About;