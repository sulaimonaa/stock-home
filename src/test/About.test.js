import React from 'react';
import { render } from '@testing-library/react';
import About from '../components/About';
import { BrowserRouter } from 'react-router-dom';

test('renders About component correctly', () => {
  const { asFragment } = render(<BrowserRouter><About /></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});
