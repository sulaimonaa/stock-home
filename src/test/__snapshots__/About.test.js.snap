// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders About component correctly 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <div
      class="container navbar"
    >
      <div
        class="home-icon"
      >
        <img
          alt="previous icon"
          src="mocked-file-url"
        />
      </div>
      <div
        class="home-icon"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <img
            alt="home icon"
            src="mocked-file-url"
          />
        </a>
      </div>
      <div
        class="home-icon"
      >
        <a
          class=""
          href="/stocks"
        >
          <img
            alt="home icon"
            src="mocked-file-url"
          />
        </a>
      </div>
      <div
        class="menu-icon"
      >
        <a
          class=""
          href="/about"
        >
          <img
            alt="menu icon"
            src="mocked-file-url"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    class="wrap"
  >
    <h3>
      About The Project
    </h3>
    <p>
      This is a simple project that uses react hook and redux to fetch financial data from an open API - Free Stock Market.
      <br />
       The home page shows the card to display financial statement list, the stock market, and about project. Credit to Nelson Sakwa on Behance for the UI layout
    </p>
    <p>
      In the financial list page, you have list of financial statements you can filter and get details of a statement by click on it. Please, note that the free subscription of the API gives limited access i.e there are some statment that will return no data.
    </p>
  </div>
</DocumentFragment>
`;
